mobile/lib/modules/home/ui/asset_list_v2/immich_asset_grid.dart

```diff
@@ -0,0 +1,166 @@
+import 'dart:math';
+
+import 'package:collection/collection.dart';
+import 'package:easy_localization/easy_localization.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/src/widgets/framework.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:immich_mobile/modules/home/providers/home_page_render_list_provider.dart';
+import 'package:immich_mobile/modules/home/ui/asset_list_v2/daily_title_text.dart';
+import 'package:immich_mobile/modules/home/ui/asset_list_v2/draggable_scrollbar_custom.dart';
+import 'package:openapi/api.dart';
+import 'package:scrollable_positioned_list/scrollable_positioned_list.dart';
+
+import '../thumbnail_image.dart';
+
+class ImmichAssetGrid extends HookConsumerWidget {
+  final ItemScrollController _itemScrollController = ItemScrollController();
+  final ItemPositionsListener _itemPositionsListener =
+      ItemPositionsListener.create();
+
+  final List<RenderAssetGridElement> renderList;
+  final int assetsPerRow;
+  final double margin;
+  final bool showStorageIndicator;
+
+  ImmichAssetGrid({
+    super.key,
+    required this.renderList,
+    required this.assetsPerRow,
+    required this.showStorageIndicator,
+    this.margin = 5.0,
+  });
+
+  List<AssetResponseDto> get _assets {
+    return renderList
+        .map((e) {
+          if (e.type == RenderAssetGridElementType.assetRow) {
+            return e.assetRow!.assets;
+          } else {
+            return List<AssetResponseDto>.empty();
+          }
+        })
+        .flattened
+        .toList();
+  }
+
+  double _getItemSize(BuildContext context) {
+    return MediaQuery.of(context).size.width / assetsPerRow -
+        margin * (assetsPerRow - 1) / assetsPerRow;
+  }
+
+  Widget _buildThumbnailOrPlaceholder(
+      AssetResponseDto asset, bool placeholder) {
+    if (placeholder) {
+      return const DecoratedBox(
+        decoration: BoxDecoration(color: Colors.grey),
+      );
+    }
+    return ThumbnailImage(
```

Comment:
I notice that the video indicator icon is not shown, I think it put it in a `Stack` widget instead of directly in the `ThumbnailImage`, it was a bad implementation in my part. Can we refactor that to be shown in `ThubmnailImage` as well?